app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,283 @@
+    button_manage: {
+        id: t('mobile.manage_members.manage'),
+        defaultMessage: 'Manage',
+    },
+    button_done: {
+        id: t('mobile.manage_members.done'),
+        defaultMessage: 'Done',
+    },
+});
+
+const MANAGE_BUTTON = 'manage-button';
+const EMPTY: UserProfile[] = [];
+const EMPTY_MEMBERS: ChannelMembership[] = [];
+const EMPTY_IDS = {};
+const {USER_PROFILE} = Screens;
+const CLOSE_BUTTON_ID = 'close-user-profile';
+
+export default function ManageChannelMembers({
+    canManageMembers,
+    channelId,
+    componentId,
+    currentTeamId,
+    currentUserId,
+    teammateNameDisplay,
+    tutorialWatched,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const {formatMessage} = useIntl();
+
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const mounted = useRef(false);
+
+    const [isManageMode, setIsManageMode] = useState(false);
+    const [profiles, setProfiles] = useState<UserProfile[]>(EMPTY);
+    const [channelMembers, setChannelMembers] = useState<ChannelMembership[]>(EMPTY_MEMBERS);
+    const [searchResults, setSearchResults] = useState<UserProfile[]>(EMPTY);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+
+    const loadedProfiles = (users: UserProfile[], members: ChannelMembership[]) => {
+        if (mounted.current) {
+            setLoading(false);
+            setProfiles(users);
+            setChannelMembers(members);
+        }
+    };
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults(EMPTY);
+    }, []);
+
+    const getProfiles = useCallback(debounce(async () => {
+        const hasTerm = Boolean(term);
+        if (!loading && !hasTerm && mounted.current) {
+            setLoading(true);
+            const {users, members} = await fetchChannelMemberships(serverUrl, channelId, true);
+            if (users.length) {
+                loadedProfiles(users, members);
+            }
+        }
+    }, 100), [channelId, loading, serverUrl, term]);
+
+    const handleSelectProfile = useCallback(async (profile: UserProfile) => {
+        if (!isManageMode) {
+            return;
+        }
+
+        const title = formatMessage({id: 'mobile.routes.user_profile', defaultMessage: 'Profile'});
+        const props = {
+            channelId,
+            closeButtonId: CLOSE_BUTTON_ID,
+            location: USER_PROFILE,
+            manageMode: true,
+            userId: profile.id,
+            canManageMembers,
+        };
+
+        Keyboard.dismiss();
+        openAsBottomSheet({screen: USER_PROFILE, title, theme, closeButtonId: CLOSE_BUTTON_ID, props});
+    }, [canManageMembers, isManageMode]);
+
+    const searchUsers = useCallback(async (searchTerm: string) => {
+        const lowerCasedTerm = searchTerm.toLowerCase();
+        setLoading(true);
+
+        // TODO: search through profiles state instead of server call
```

Comment:
This makes sense, but we would need to duplicate how the API uses search based on the search term.

Reply:
Ticket number if needed else remove comment